src/Illuminate/Database/Schema/Blueprint.php

```diff
@@ -678,6 +678,37 @@ public function index($columns, $name = null, $algorithm = null)
         return $this->indexCommand('index', $columns, $name, $algorithm);
     }
 
+    /**
+     * Specify a functional index for the table.
+     *
+     * This method allows you to create a functional index using MySQL expressions
+     * like `LOWER`, `UPPER`, or other supported functions directly on columns.
+     *
+     * @param  string|array  $columns  The column(s) to include in the index.
+     * @param  string $function The SQL function to apply to the column(s).
+     * @param string $name The name of index if you want to add a unique index
+     * @return \Illuminate\Database\Schema\IndexDefinition
+     */
+    public function functionalIndex($columns, $function, $name = null)
+    {
+        $columns = is_array($columns) ? $columns : [$columns];
```

Comment:
This should suffice:
```suggestion
        $columns = (array)$columns;
```